### DISCUSSION TITLE
Unable to upload to a MKS DLC 2.0 clone (Atomstack A5 laser)

### DISCUSSION THREAD

**Discussion by mldmld68** [Optiboot/optiboot#310] [Usage problems]
Hi,
I bought an Atomstack A5 laser and added limit switches.  This machine is build with a Makerbase MKS DLC 2.0 clone (LaserBox V1.0).
See https://github.com/makerbase-mks/MKS-DLC
See https://fr.banggood.com/mypage-index-Qc+GvH3a48FZrSM0hKB78g.html (I want to do a similar setting)

It has ATMEGA328P-AU, 8bit 20MHZ
I installed Teraterm and using 115200 bauds, USB-Serial CH340 (COM4)
Communication is ok as I can get the values with $$ 

I noticed
- It is provided with GRBL 1.1f  and the variable settings changes are working well but thet are NOT retained in the EEPROM after power cycle.
I read this writing limitation could be related to the boot firmware.
That's why I'm trying to install Optiboot and later GRBL 1.1h if required.

To do so, I installed Arduino IDE 1.8.13 and I configured the new MiniCore board
Then
I closed Arduino IDE. I plug the USB cable with the board
I selected 
- Tools > Board >  atmega328
- Tools > Clock: External 20 Mhz
- Tools > Variants: 328P
I don't have any programmer. Just an USB cable

- Tools > GetBoard info returns 
   BN: Unknown board
   VID: 1A86
   PID: 7523
   SN: Upload any sketch to obtain it

Tools > Serial Monitor works when I setup baudrate to 115200.

While trying Burn Bootloader, I get 
Arduino: 1.8.13 (Windows 10), Board: "ATmega328, Yes (UART0), EEPROM retained, 328P / 328PA, BOD 2.7V, LTO disabled, External 20 MHz"
C:\Users\Marc\AppData\Local\Arduino15\packages\arduino\tools\avrdude\6.3.0-arduino18/bin/avrdude -CC:\Users\Marc\AppData\Local\Arduino15\packages\MiniCore\hardware\avr\2.1.1/avrdude.conf -v -patmega328p -cstk500v2 -Pusb -e -Ulock:w:0x3f:m -Uefuse:w:0b11111101:m -Uhfuse:w:0b11010110:m -Ulfuse:w:0b11110111:m 
avrdude: Version 6.3-20201216
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch
         System wide configuration file is "C:\Users\Marc\AppData\Local\Arduino15\packages\MiniCore\hardware\avr\2.1.1/avrdude.conf"
        Using Port                    : usb
        Using Programmer              : stk500v2
avrdude: usbdev_open(): did not find any USB device "usb" (0x03eb:0x2104)
avrdude done.  Thank you.
Error while burning bootloader.

How can I make things work ? I don't see an option with "No Programmer" in Arduino IDE.

BTW, will I need to flash the GRBL 1.1h too ? Or the Bootloader firmware and GRBL are independants ?

Thank you
MLD

**Comment by WestfW** [comment]
> Tools > Variants: 328P
> I don't have any programmer. Just an USB cable

You must have a programmer of some kind to burn or replace the bootloader.

EEPROM functionality should not depend at all on the bootloader; requiring a new bootloader would only be necessary if GRBL was trying to save the settings in flash, rather than in EEPROM (which would be "unlikely" on an ATmega328, I think.)
